### PR TITLE
Added caching for Volume info, Peers and IsLeader functions

### PR DESCRIPTION
**Pull request by aravindavk**
Fixes: #125
Signed-off-by: Aravinda VK <avishwan@redhat.com>

### CODE REVIEW

File: extras/conf/gluster-exporter.toml.sample [hunk 1]
```diff
@@ -8,6 +8,7 @@ metrics-path = "/metrics"
 log-dir = "/var/log/gluster-exporter"
 log-file = "exporter.log"
 log-level = "info"
+cache-ttl = 7
```
Comment (shtripat) [inline extras/conf/gluster-exporter.toml.sample]: Is 7 sec good enough? Any logic behind this? Can we make this 10 sec?

**Comment by shtripat** [comment]
@sidharthanup I understand the point here and would leave it to @aravindavk as if there are some perf related overload due to this.

**Comment by aravindavk** [comment]
@sidharthanup since read is from memory so it will be fraction of second, takes time only when cache is expired.

**Comment by aravindavk** [comment]
@aruniiird is working on an interesting approach with less code duplication for caching. Will wait for his PR.

**Comment by shtripat** [comment]
@aravindavk I would suggest to take this PR in at the moment and later once all the required changes done and verified in https://github.com/gluster/gluster-prometheus/pull/133 we would take that.
@aravindavk can we merge this now?

**Comment by aravindavk** [comment]
> @aravindavk I would suggest to take this PR in at the moment and later once all the required changes done and verified in #133 we would take that.
@aravindavk can we merge this now?

Added one minor comment to #133, we can merge that PR and add any enhancements later. Good thing in that PR is collectors need not change anything to get cached output. Also in future if caching needs to be disabled that can be done with a config change.(Code change required in my PR)

**Pull request closed without merging**